OpenEdXMobile/src/main/java/org/edx/mobile/view/LoginActivity.java

```diff
@@ -166,14 +167,19 @@ protected void onRestoreInstanceState(Bundle savedInstanceState) {
     }
 
     @Override
-    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
         tryToSetUIInteraction(true);
         socialLoginDelegate.onActivityResult(requestCode, resultCode, data);
 
-        if (resultCode == ResetPasswordDialogFragment.RESULT_OK) {
-            showAlertDialog(getString(R.string.success_dialog_title_help),
-                    getString(R.string.success_dialog_message_help));
+        switch (requestCode) {
+            case ResetPasswordDialogFragment.REQUEST_CODE: {
+                if (resultCode == Activity.RESULT_OK) {
```

Comment:
It would be consistent to use `switch` for this too, although I suppose if we're only ever thinking of checking this value then using `if` is fine as well.